### PR TITLE
Reject supplied nonces for non-convergent encryption operations

### PR DESCRIPTION
**Pull request by sgmiller**
Fixes a regression where nonces were being honored if provided on encrypt
operations in non-convergent keys.

### CODE REVIEW

File: builtin/logical/transit/path_rewrap.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/mitchellh/mapstructure"
 )
 
+var ErrNonceNotAllowed = errors.New("provided nonce not allowed for this key")
```
Comment (schultz-is) [inline builtin/logical/transit/path_rewrap.go:19]: looks like there's an import missing of the `errors` package

File: builtin/logical/transit/path_encrypt.go [hunk 2]
```diff
@@ -568,6 +573,23 @@ func (b *backend) pathEncryptWrite(ctx context.Context, req *logical.Request, d
 	return batchRequestResponse(d, resp, req, successesInBatch, userErrorInBatch, internalErrorInBatch)
 }
 
+func nonceAllowed(p *keysutil.Policy) bool {
```
Comment (stevendpclark) [inline builtin/logical/transit/path_encrypt.go:576]: Do we need to consider managed keys here?

Comment (sgmiller) [reply builtin/logical/transit/path_encrypt.go:578]: Hmm, with managed keys we don't know if a nonce is needed or not, right?

Comment (stevendpclark) [reply builtin/logical/transit/path_encrypt.go:578]: I believe that is correct, but with this it will never make it through.

Comment (schultz-is) [reply builtin/logical/transit/path_encrypt.go:578]: correct. especially in PKCS#11, it depends on the mechanism used.

**Review by cipherboy — APPROVED** [review]
Based on internal discussion, I believe this looks good!

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**